ports/stm/common-hal/busio/UART.c

```diff
@@ -313,6 +336,10 @@ void HAL_UART_RxCpltCallback(UART_HandleTypeDef *handle)
             }
             ringbuf_put_n(&context->ringbuf, &context->rx_char, 1);
             errflag = HAL_UART_Receive_IT(handle, &context->rx_char, 1);
+            if (context->rx_char == CHAR_CTRL_C) {
+                common_hal_busio_uart_clear_rx_buffer(context);
+                mp_keyboard_interrupt();
```

Comment:
We shouldn't always do this. The uart should have a setting for it.

Someone may want full control of a UART and not want it do detect ctrl-c.

Reply:
I've added a new argument to the construct to decide if we listen for sigint (ctrl-c). I would like to add some callback support to uart, so it's easier to add multiple listeners, but that's out of the scope of this issue I think.